src/lib.rs

```diff
@@ -829,30 +839,29 @@ impl AcquireState {
     pub fn complete(&self) -> &AtomicBool {
         // Safety: This is always safe to access since it's atomic.
         unsafe {
-            let ptr = self.linking.get() as *const _ as *const Node<Task>;
-            let ptr = ptr.add(1) as *const AtomicBool;
+            let ptr = addr_of!((*self.linking.get()).complete);
             &*ptr
         }
     }
 
     /// Get the underlying task.
     pub unsafe fn task(&self) -> &Node<Task> {
-        let ptr = self.linking.get() as *mut Node<Task>;
-        &*ptr
+        unsafe {
+            let ptr = addr_of!((*self.linking.get()).task);
+            &*ptr
+        }
     }
 
     /// Get the underlying task mutably.
-    pub unsafe fn task_mut(&mut self) -> &mut Node<Task> {
-        let ptr = self.linking.get() as *mut Node<Task>;
-        &mut *ptr
+    pub fn task_mut(&mut self) -> &mut Node<Task> {
+        &mut self.linking.get_mut().task
```

Comment:
This and the other safe use of `get_mut()` looks suspicious to me. What I'd be worried about is if it might end up violating stacked borrows with or without strict provenance.

i.e. if we project out through a borrowed reference of `linked`, the project borrow owns what it's been project out of uniquely (in this case maybe `&mut self`?), but we might also project out task or related fields elsewhere.

I would be more comfortable if we kept these as raw pointer projections.